server/server.go

```diff
@@ -869,36 +769,25 @@ func (s *server) RestartNode(ctx context.Context, req *rpcpb.RestartNodeRequest)
 		return nil, err
 	}
 
-	if err := s.network.updateNodeInfo(); err != nil {
+	if err := s.network.UpdateNodeInfo(); err != nil {
 		return nil, err
 	}
 
-	s.clusterInfo.NodeNames = s.network.nodeNames
+	s.clusterInfo.NodeNames = maps.Keys(s.network.nodeInfos)
 	s.clusterInfo.NodeInfos = s.network.nodeInfos
 
 	return &rpcpb.RestartNodeResponse{ClusterInfo: s.clusterInfo}, nil
 }
 
-func (s *server) Stop(ctx context.Context, _ *rpcpb.StopRequest) (*rpcpb.StopResponse, error) {
-	s.log.Debug("Stop")
-
-	if s.getNetwork() == nil {
-		return nil, ErrNotBootstrapped
-	}
-
+func (s *server) Stop(context.Context, *rpcpb.StopRequest) (*rpcpb.StopResponse, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	s.network.stop(ctx)
-	s.network = nil
+	s.log.Debug("Stop")
 
-	info := s.clusterInfo
-	if info == nil {
-		info = &rpcpb.ClusterInfo{}
-	}
-	info.Healthy = false
+	s.stopAndRemoveNetwork()
 
-	return &rpcpb.StopResponse{ClusterInfo: info}, nil
+	return &rpcpb.StopResponse{}, nil
```

Comment:
previously this was returning latest valid cluster info

Reply:
The clusterInfo will always be empty so I don't see a need to return it

Reply:
clusterinfo was being returned on stop, pls try to keep the same semantics